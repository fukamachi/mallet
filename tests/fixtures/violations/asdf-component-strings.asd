;; Test fixtures for ASDF component strings rule

;; Bad: system name as symbol
(defsystem #:my-project
  :depends-on ("alexandria"))

;; Good: system name as string
(defsystem "my-project-good"
  :depends-on ("alexandria"))

;; Bad: dependencies as symbols
(defsystem "another-project"
  :depends-on (#:alexandria
               #:cl-ppcre))

;; Good: dependencies as strings
(defsystem "another-project-good"
  :depends-on ("alexandria"
               "cl-ppcre"))

;; Bad: component names as symbols
(defsystem "project-with-components"
  :components ((:file #:main)
               (:file #:utils)))

;; Good: component names as strings
(defsystem "project-with-components-good"
  :components ((:file "main")
               (:file "utils")))

;; Bad: mixed - some symbols, some strings
(defsystem #:mixed-project
  :depends-on (#:alexandria
               "cl-ppcre")
  :components ((:file #:main)
               (:file "utils")))
