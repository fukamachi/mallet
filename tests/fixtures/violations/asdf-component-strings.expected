# Expected violations for asdf-component-strings.asd
# Format: line:column rule-name severity
# Note: All violations report at form start line, not precise symbol position

4:0 asdf-component-strings convention
12:0 asdf-component-strings convention
12:0 asdf-component-strings convention
22:0 asdf-component-strings convention
22:0 asdf-component-strings convention
32:0 asdf-component-strings convention
32:0 asdf-component-strings convention
32:0 asdf-component-strings convention
